extension-style/src/main/java/com/mapbox/maps/extension/style/sources/generated/GeoJsonSource.kt

```diff
@@ -618,30 +699,53 @@ class GeoJsonSource(builder: Builder) : Source(builder.sourceId) {
      * @param value the geometry
      */
     fun geometry(value: Geometry) = apply {
-      rawGeoJson = value
-      val propertyValue = PropertyValue("data", "null")
-      properties[propertyValue.propertyName] = propertyValue
+      geoJson = value
+      if (directSetterEnabled()) {
+        data = null
+      } else {
+        val propertyValue = PropertyValue("data", "null")
+        properties[propertyValue.propertyName] = propertyValue
+      }
     }
 
     /**
      * Initialize GeoJsonSource builder with default data to allow empty data source.
      */
     init {
-      this.data("")
+      if (!directSetterEnabled()) {
+        this.data("")
+      }
     }
 
     /**
      * Build the GeoJsonSource.
      *
      * @return the GeoJsonSource
      */
-    fun build() = GeoJsonSource(this, rawGeoJson)
+    fun build(): GeoJsonSource {
+      if (directSetterEnabled()) {
+        // we need empty initial data when creating the geojson (as we don't set it in constructor) -
+        // actual data (if specified by the user) will be applied asynchronously later
+        val propertyValue = PropertyValue("data", TypeUtils.wrapToValue(""))
+        properties[propertyValue.propertyName] = propertyValue
+      }
+      return GeoJsonSource(this, geoJson, data)
+    }
   }
 
   /**
    * Static variables and methods.
    */
   companion object {
+    private const val TAG = "GeoJsonSource"
+
+    private fun directSetterEnabled(): Boolean {
+      val settingValue = SettingsServiceFactory
+        .getInstance(SettingsServiceStorageType.NON_PERSISTENT)
+        .get("geojson_allow_direct_setter")
```

Comment:
if it possible it will be nice to save to some field and don't recalculate it each time

Reply:
fixed